### DISCUSSION TITLE
Private Registry registration is not working.

### DISCUSSION THREAD

**Discussion by Limseongbin** [rancher/rke2#7074] [Q&A] [answered]
I created the /etc/rancher/rke2/registries.yaml file to register a private registry.

Restarting the rke2-server kre2-agent service does not synchronize the registry information in the /var/lib/rancher/rke2/agent/etc/containerd/config.toml file.

It works fine in v1.26.10+rke2r2. However, it doesn't work in v.1.26.15+rke2r1 and v1.30.5+rke2r1.
I haven't tested all versions... but at least the latest version is having issues

**Comment by brandond** [comment, marked as answer]
Placing registry configuration in config.toml has long been deprecated by containerd, in favor of a dynamically loaded certs.d directory structure. This is what newer releases of RKE2 use; you'll find the registry config under `/var/lib/rancher/rke2/agent/etc/containerd/certs.d`.

Ref: https://github.com/k3s-io/k3s/issues/8972

**Reply by Limseongbin** [reply]
Ok, Thanks.
